AArch64: Use RuntimeLibcallsInfo in SMEAttributes (NFC)

arsenm opened the pull request:
Eventually this should be program state, and not part of TargetLowering
so avoid direct references to the libcall functions in it.

The usage of RuntimeLibcallsInfo here is not good though, particularly
the use through TargetTransformInfo. It would be better if the IR attributes
were directly encoded in the libcall definition (or at least made consistent
elsewhere). The parsing of the attributes should not also be responsible for
doing the libcall recognition, which is the only part pulling in the
dependency.

sdesmalen-arm reviewed (approved):
I think this can have `(NFC)` in the title?

arsenm committed:
AArch64: Use RuntimeLibcallsInfo in SMEAttributes (NFC)

Eventually this should be program state, and not part of TargetLowering
so avoid direct references to the libcall functions in it.

The usage of RuntimeLibcallsInfo here is not good though, particularly
the use through TargetTransformInfo. It would be better if the IR attributes
were directly encoded in the libcall definition (or at least made consistent
elsewhere). The parsing of the attributes should not also be responsible for
doing the libcall recognition, which is the only part pulling in the
dependency.

MacDue commented on llvm/lib/Target/AArch64/Utils/AArch64SMEAttributes.h:
```diff
@@ -12,6 +12,9 @@
+struct RuntimeLibcallsInfo;
+}
 
 class AArch64TargetLowering;
```
Remove the `AArch64TargetLowering` forward decl (and I think the include of `AArch64ISelLowering.h` in the C++ file can be replaced).

arsenm committed:
Remove forward decl
